### ISSUE TITLE
`no-unused-expressions` false positive ?

### ISSUE THREAD

**Issue by romaindurand** [eslint/eslint#8282]
**Tell us about your environment**

* **ESLint Version: eslint@3.18.0**
* **Node Version: v7.7.2**
* **npm Version: 4.1.2**
* **VSCode Version: 1.10.2**
* @feross **standard Version: 9.0.2 (also tried with 10.0.0-beta.2)**
* my VSCode extension for standard uses my global installation of standard

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** feross/standard
No unusual configuration, I'm using @feross 's standard preset

**What did you do? Please include the actual source code causing the issue.**
>test.js
```js

const errors = []
errors.push['test']

console.log(errors)

```

**What did you expect to happen?**
I think that I shouldn't get an error with this code

**What actually happened? Please include the actual, raw output from ESLint.**
>\> standard test.js
standard: Use JavaScript Standard Style (http://standardjs.com)
test.js:2:1: Expected an assignment or function call and instead saw an expression.

**Comment by ilyavolodin** [comment]
The rule is correctly flagging second line, it should be this instead:
```js
errors.push('test');
```

**Comment by romaindurand** [comment]
I really should go to sleep, thanks for the help, and sorry for the noise.
